src/Propel/Runtime/ActiveQuery/ModelCriteria.php

```diff
@@ -2013,23 +2013,24 @@ protected function doReplaceNameInExpression($matches)
         $key = $matches[0];
         [$column, $realFullColumnName] = $this->getColumnFromName($key);
 
-        if ($column instanceof ColumnMap) {
-            $this->replacedColumns[] = $column;
-            $this->foundMatch = true;
+        if (!$column instanceof ColumnMap) {
```

Comment:
Couldn't read the method, so I untangled it. Again, not really part of the bugfix. Left it in because it is less convoluted.